src/Intersection.jl

```diff
@@ -370,16 +433,30 @@ Return if the intersection is empty or not.
 ### Output
 
 `true` iff the intersection is empty.
+
+### Notes
+
+The result will be cached, so a second query will be fast.
 """
 function isempty(cap::Intersection)::Bool
-    return is_intersection_empty(cap.X, cap.Y)
+    if isempty_known(cap.cache)
```

Comment:
above there is a method `isempty_konwn(::Intersection)`. should we use it instead? (or delete it?)